internal/publisher/kafka.go

```diff
@@ -41,10 +62,24 @@ func (p *KafkaPublisher) Close() error {
 	return p.producer.Close()
 }
 
+func (p *KafkaPublisher) keyData(e *Event) []byte {
+	if data, ok := e.Data[p.cfg.MessageKeyFrom]; ok {
```

Comment:
I think it is important to handle the `DELETE` case here and to check for the presence of the key in `e.DataOld` map so that the complete lifecycle of the data is in the same partition (CREATE, UPDATE, UPDATE, ..., DELETE). TBH I missed it in my first implementation